tests/lib/rules/no-extra-parens.js

```diff
@@ -129,8 +129,14 @@ ruleTester.run("no-extra-parens", rule, {
         "[a, b]",
         "!{a: 0, b: 1}",
 
+        // ExpressionStatement restricted productions
+        "({});",
+        "(function(){});",
+        "(let[a] = b);",
```

Comment:
I couldn't add this same test under ES6 mode because of this outstanding espree issue: https://github.com/eslint/espree/issues/149